src/components/UserSelector.tsx

```diff
@@ -2,39 +1,97 @@
+}) => {
+  const [showUsers, setShowUsers] = useState(false);
+  const dropdownMenuRef = useRef<HTMLDivElement | null>(null);
+  const dropdownTriggerRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownMenuRef.current &&
+        !dropdownMenuRef.current.contains(event.target as Node) &&
+        dropdownTriggerRef.current &&
+        !dropdownTriggerRef.current.contains(event.target as Node)
+      ) {
+        setShowUsers(false);
+      }
+    };
+
+    document.addEventListener('click', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, []);
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
-      <div className="dropdown-trigger">
+    <div
+      data-cy="UserSelector"
+      className={classNames('dropdown', { 'is-active': showUsers })}
+    >
+      <div className="dropdown-trigger" ref={dropdownTriggerRef}>
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setShowUsers(prev => !prev)}
         >
-          <span>Choose a user</span>
+          {userSelected ? (
+            <span>{userSelected.name}</span>
+          ) : (
+            <span>Choose a user</span>
+          )}
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
-        <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">
-            Leanne Graham
-          </a>
-          <a href="#user-2" className="dropdown-item is-active">
-            Ervin Howell
-          </a>
-          <a href="#user-3" className="dropdown-item">
-            Clementine Bauch
-          </a>
-          <a href="#user-4" className="dropdown-item">
-            Patricia Lebsack
-          </a>
-          <a href="#user-5" className="dropdown-item">
-            Chelsey Dietrich
-          </a>
+      <div
+        className="dropdown-menu"
+        id="dropdown-menu"
+        role="menu"
+        ref={dropdownMenuRef}
+      >
+        <div
+          className="dropdown-content"
+          style={{ display: showUsers ? 'block' : 'none' }}
+        >
+          {users.map(user => (
+            <a
+              href={`#user-${user.id}`}
+              className={classNames('dropdown-item', {
+                'is-active': user.id === userSelected?.id,
+              })}
+              key={user.id}
+              onClick={() => {
+                setUserSelected(user);
+                setShowUsers(false);
+                handleUserPosts(user);
+                setShowSideBar(null);
+              }}
```

Comment:
You are calling setState multiple times in one function call. It is recommended to avoid this for clarity of the code. This is a code quality issue.